File: rust/project_012/main.rs
use std::io;

fn main() {
    println!("Enter a string to reverse:");
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read");
    let s = input.trim();
    let reversed: String = s.chars().rev().collect();
    println!("Reversed: {}", reversed);
}

File: typescript/project_012/main.ts
// Project 012: Fahrenheit to Celsius Converter CLI
import * as readline from 'readline';

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('Enter temperature in Fahrenheit: ', (f) => {
    const fahrenheit = Number(f);
    if (isNaN(fahrenheit)) {
        console.log('Invalid input.');
    } else {
        const celsius = ((fahrenheit - 32) * 5) / 9;
        console.log(`${fahrenheit}°F = ${celsius.toFixed(2)}°C`);
    }
    rl.close();
});
